ui/app/components/feedback/FeedbackTableModal.tsx

```diff
@@ -42,17 +46,26 @@ export function DemonstrationModal({ feedback }: FeedbackTableModalProps) {
     return <div>Invalid demonstration feedback</div>;
   }
 
+  // Try to parse the demonstration value as JSON for the parsed property
+  let parsedValue = null;
+  try {
+    parsedValue = JSON.parse(feedback.value);
+  } catch {
+    // If parsing fails, keep parsedValue as null
```

Comment:
Is non-JSON actually expected here in some cases? If not, should we capture the error somehow, or is showing the raw value good enough?

Reply:
@maxbokov I think this is actually incorrect generally. The demonstration data should parse to a JSON that matches `JsonInferenceOutput | ContentBlockOutput[]` so if it doesn't parse there then this is an error and we should render that error to the user in a loud way. The correct way to do this is to use `parseInferenceOutput` in `inference.ts` and handle an error if it throws one. Sorry for not being clearer here. But then, you'll have the corrrect type for rendering this with an `Output` or `NewOutput` component, as desired.